packages/speech/src/helpers.js

```diff
@@ -0,0 +1,120 @@
+ * limitations under the License.
+ */
+
+/*!
+ * @module speech/helpers
+ */
+
+'use strict';
+
+var pumpify = require('pumpify');
+var streamEvents = require('stream-events');
+var through = require('through2');
+
+/*!
+ * Return a dictionary-like object with helpers to augment the Speech
+ * GAPIC.
+ *
+ * @return {Object} - An object with keys and functions which are placed
+ *   onto the pure GAPIC.
+ */
+module.exports = () => {
+  var methods = {};
+
+  /**
+   * Performs bidirectional streaming speech recognition: receive results while
+   * sending audio. This method is only available via the gRPC API (not REST).
+   *
+   * @param {Object} config
+   *   The configuration for the stream. This is appropriately wrapped and
+   *   sent as the first argument. It should be an object conforming to the
+   *   [StreamingRecognitionConfig]{@link StreamingRecognitionConfig}
+   *   structure.
+   * @param {Object=} options
+   *   Optional parameters. You can override the default settings for this
+   *   call, e.g, timeout, retries, paginations, etc. See
+   *   [gax.CallOptions]{@link https://googleapis.github.io/gax-nodejs/global.html#CallOptions}
+   *   for the details.
+   * @returns {Stream}
+   *   An object stream which is both readable and writable. It accepts
+   *   [StreamingRecognizeRequest]{@link StreamingRecognizeRequest}-like
+   *   objects for the write() method, and will emit objects representing
+   *   [StreamingRecognizeResponse]{@link StreamingRecognizeResponse} on the
+   *   'data' event asynchronously.
+   *
+   * @example
+   *
+   * var client = require('@google-cloud/speech')();
+   *
+   * var stream = client.streamingRecognize({
+   *   config: {
+   *     encoding: 'LINEAR16',
+   *     languageCode: 'en-us',
+   *     sampleRateHertz: 44100,
+   *   },
+   * }).on('data', function(response) {
+   *   // doThingsWith(response);
+   * });
+   * var request = {};
+   * // Write request objects.
+   * stream.write(request);
+   */
+  methods.streamingRecognize = function(config, options) {
+    if (options === undefined) {
+      options = {};
+    }
+
+    var requestStream = this._streamingRecognize(options);
+
+    // Format the audio content as input request for pipeline
+    var recognizeStream = streamEvents(pumpify.obj());
+
+    recognizeStream.once('writing', function() {
+      requestStream.on('error', function(err) {
+        recognizeStream.destroy(err);
+      });
+
+      requestStream.on('response', function(response) {
+        recognizeStream.emit('response', response);
+      });
+
+      // Write the initial configuration to the stream,
+      requestStream.write({
+        streamingConfig: config
+      });
+
+      this.setPipeline([
+        // Format the user's input.
+        through.obj(function(obj, _, next) {
+          next(null, {
+            audioContent: obj
+          });
+        }),
+
+        requestStream,
+
+        // Pass through the results.
```

Comment:
We should be able to remove this part. The following stream:

```js
through.obj(function(obj, _, next) {
  next(null, obj);
})
```

is the same as just:

```js
through.obj();
```

which is just a passthrough stream, which means we might as well save the memory (streams are relatively heavy) and not use it at all.